src/unix/tcp.c

```diff
@@ -33,7 +33,25 @@ static int maybe_new_socket(uv_tcp_t* handle, int domain, int flags) {
   int err;
 
   if (domain == AF_UNSPEC || uv__stream_fd(handle) != -1) {
+    struct sockaddr_storage saddr;
+    socklen_t slen;
+
     handle->flags |= flags;
+
+    slen = sizeof(saddr);
+    if (getsockname(uv__stream_fd(handle), (struct sockaddr*) &saddr, &slen))
+      return -errno;
+
+    /* If the handle is already bound to a port, flag
+     * it as bound.
+     */
+    if (saddr.ss_family == AF_INET6 &&
+       ((struct sockaddr_in6*) &saddr)->sin6_port != 0)
+      handle->flags |= UV_HANDLE_BOUND;
+    else if (saddr.ss_family == AF_INET &&
+            ((struct sockaddr_in*) &saddr)->sin_port != 0)
+      handle->flags |= UV_HANDLE_BOUND;
+
```

Comment:
I'm not too sure about this. I think `UV_HANDLE_BOUND` is not set in the handle after `uv_connect()` succeeds, but it looks like this could set `UV_HANDLE_BOUND` if the passed socket is a connected socket. Couldn't this be a problem?